POLIO-1451, POLIO-1464: refactor PreAlert, ArrivalReport

quang-le opened the pull request:
Explain what problem this PR is resolving

Related JIRA tickets : POLIO-1451, POLIO-1464

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc
NA

## Changes

- Marked all fields that might crash the API or the front-end if null as required in the interface and the yup validation
- Remove lot numbers and expiry date from PreAlert and ArrivalReport.

## How to test
In package.json, change the dep to bluesquare-components to :
` "bluesquare-components": "github:BLSQ/bluesquare-components#IA-2681_date_picker_lost_props",`

and update you webpack build (docker) or run `npm update bluesquare-components`.

Then: 

Go to supplychain, try to create/edit VRF, PreAlert and Arrival reports

## Print screen / video
![Screenshot 2024-02-05 at 16 59 12](https://github.com/BLSQ/iaso/assets/38907762/cdbb74e4-1c5f-456d-9944-8f258c3cc72a)

![Screenshot 2024-02-05 at 16 58 47](https://github.com/BLSQ/iaso/assets/38907762/b9af4616-07d1-44f0-b525-70bfce71d258)


## Notes

Requires [IA-2681_date_picker_lost_props](https://github.com/BLSQ/bluesquare-components/pull/142) on bluesquare-components for the `DatePicker` to properly appears `required` and to have its label with the right color when errored --> DONE

quang-le committed:
POLIO-1451: mark mandatory fields

quang-le committed:
Merge branch 'main' into POLIO-1451_mrk-mandatory_fields

quang-le committed:
POLIO-1451: update required fields in validation (supply chain)

quang-le committed:
POLIO-1464: remove lot nr and exp date (suppplychain)

- In PreAlert
- In ArrivalReport
- In yup validation

quang-le committed:
POLIO-1451: update bluesquare-components

quang-le committed:
Merge remote-tracking branch 'origin/main' into POLIO-1451_mrk-mandatory_fields

beygorghor commented on plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx:
```diff
@@ -212,7 +214,7 @@ export const VaccineRequestForm: FunctionComponent<Props> = ({
                                 component={NumberInput}
                                 disabled={false}
-                                required
+                                // required
```
🗑️

beygorghor commented on plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx:
```diff
@@ -245,7 +247,7 @@ export const VaccineRequestForm: FunctionComponent<Props> = ({
                                 component={NumberInput}
                                 disabled={false}
-                                required
+                                // required
```
🗑️

beygorghor reviewed (changes requested):
Can we add a * to display required date fields like other inputs ?

I also have this issue while trying to create a VRF, but this is probably due to the quality of my data


https://github.com/BLSQ/iaso/assets/12494624/16ebd7c2-3fbc-48b4-86c2-78fb41bdb14e

quang-le committed:
Merge branch 'main' into POLIO-1451_mrk-mandatory_fields

beygorghor commented:
My bad, required fields are displayed, remains other bug and while editing a VRF
Also i would disable cancel button while editing a VRF and nothing changed

quang-le committed:
POLIO-1451: add better translation for cancel button

- Cancel changes

quang-le committed:
POLIO-1451: disable cancel button if no changes in the form

quang-le committed:
POLIO-1451: fix bug when changing vaccines

quang-le committed:
POLIO-1451: improve allowConfirm condition

- Remove check on touched, since we tell if a form has been changed by comparing values and initialValues
- pass wastage ratio to parseFloat when setting initialValues to match the behaviour NumberInput, which will convert the string value to float
